--- a/spipe/main.c
+++ b/spipe/main.c
@@ -71,7 +71,11 @@ callback_graceful_shutdown(void * cookie
 
 	/* Cancel the threads. */
 	for (i = 0; i < 2; i++) {
+#ifndef __ANDROID__
 		if ((rc = pthread_cancel(ET->threads[i])) != 0) {
+#else
+		if ((rc = pthread_kill(ET->threads[i], SIGUSR2)) != 0) {
+#endif
 			/*
 			 * According to the POSIX standard, a Thread ID should
 			 * still be valid after pthread_exit has been invoked
@@ -310,12 +314,20 @@ main(int argc, char * argv[])
 	exit(0);
 
 err6:
+#ifndef __ANDROID__
 	if ((rc = pthread_cancel(ET.threads[0])) != 0)
+#else
+	if ((rc = pthread_kill(ET.threads[0], SIGUSR2)) != 0)
+#endif
 		warn0("pthread_cancel: %s", strerror(rc));
 	if ((rc = pthread_join(ET.threads[0], NULL)) != 0)
 		warn0("pthread_join: %s", strerror(rc));
 err5:
+#ifndef __ANDROID__
 	if ((rc = pthread_cancel(ET.threads[1])) != 0)
+#else
+	if ((rc = pthread_kill(ET.threads[1], SIGUSR2)) != 0)
+#endif
 		warn0("pthread_cancel: %s", strerror(rc));
 	if ((rc = pthread_join(ET.threads[1], NULL)) != 0)
 		warn0("pthread_join: %s", strerror(rc));
