diff -u -r ../termux-api-package-0.58.0/termux-api.c ./termux-api.c
--- ../termux-api-package-0.58.0/termux-api.c	2023-12-23 20:51:12.000000000 +0000
+++ ./termux-api.c	2024-07-03 07:51:23.833929298 +0000
@@ -60,7 +60,7 @@
     // the app process to deliver the intent.
     // - https://github.com/termux/termux-api/issues/638#issuecomment-1813233924
     int listenfd = -1;
-    #ifdef __ANDROID__
+    #ifdef __ANDROID_DISABLED__
         if (android_get_device_api_level() < 34) {
             listenfd = socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC, 0);
         }
@@ -246,15 +246,15 @@
     // Close stdin:
     close(STDIN_FILENO);
 
-    int const extra_args = 15; // Including ending NULL.
+    int const extra_args = 17; // Including ending NULL.
     char** child_argv = malloc((sizeof(char*)) * (argc + extra_args));
 
     child_argv[0] = "am";
-    child_argv[1] = "broadcast";
+    child_argv[1] = "startservice";
     child_argv[2] = "--user";
     child_argv[3] = "0";
     child_argv[4] = "-n";
-    child_argv[5] = "com.termux.api/.TermuxApiReceiver";
+    child_argv[5] = "com.termux/com.termux.app.TermuxService";
     child_argv[6] = "--es";
     // Input/output are reversed for the java process (our output is its input):
     child_argv[7] = "socket_input";
@@ -265,6 +265,8 @@
     child_argv[12] = "--es";
     child_argv[13] = "api_method";
     child_argv[14] = argv[1];
+    child_argv[15] = "-a";
+    child_argv[16] = "com.termux.service_api";
 
     // Copy the remaining arguments -2 for first binary and second api name:
     memcpy(child_argv + extra_args, argv + 2, (argc-1) * sizeof(char*));
