diff -u -r ../ncurses-snapshots-801929d34ab7fda3ca77ef59484b34a127adf8d3/ncurses/tinfo/tinfo_driver.c ./ncurses/tinfo/tinfo_driver.c
--- ../ncurses-snapshots-801929d34ab7fda3ca77ef59484b34a127adf8d3/ncurses/tinfo/tinfo_driver.c	2024-05-19 23:41:13.000000000 +0000
+++ ./ncurses/tinfo/tinfo_driver.c	2024-05-24 09:08:53.941953063 +0000
@@ -496,21 +496,6 @@
 		    _nc_setenv_num("COLUMNS", *colp);
 		}
 	    }
-
-	    /*
-	     * Finally, look for environment variables.
-	     *
-	     * Solaris lets users override either dimension with an environment
-	     * variable.
-	     */
-	    if ((value = _nc_getenv_num("LINES")) > 0) {
-		*linep = Min(value, MAX_ENV_LINES);
-		T(("screen size: environment LINES = %d", *linep));
-	    }
-	    if ((value = _nc_getenv_num("COLUMNS")) > 0) {
-		*colp = Min(value, MAX_ENV_COLUMNS);
-		T(("screen size: environment COLUMNS = %d", *colp));
-	    }
 	}
 
 	/* if we can't get dynamic info about the size, use static */
