diff -u -r ../ncurses-snapshots-801929d34ab7fda3ca77ef59484b34a127adf8d3/ncurses/tinfo/lib_setup.c ./ncurses/tinfo/lib_setup.c
--- ../ncurses-snapshots-801929d34ab7fda3ca77ef59484b34a127adf8d3/ncurses/tinfo/lib_setup.c	2024-05-19 23:41:13.000000000 +0000
+++ ./ncurses/tinfo/lib_setup.c	2024-05-24 09:07:44.008125540 +0000
@@ -562,21 +562,6 @@
 		}
 	    }
 
-	    /*
-	     * Finally, look for environment variables.
-	     *
-	     * Solaris lets users override either dimension with an environment
-	     * variable.
-	     */
-	    if ((value = _nc_getenv_num("LINES")) > 0) {
-		*linep = Min(value, MAX_ENV_LINES);
-		T(("screen size: environment LINES = %d", *linep));
-	    }
-	    if ((value = _nc_getenv_num("COLUMNS")) > 0) {
-		*colp = Min(value, MAX_ENV_COLUMNS);
-		T(("screen size: environment COLUMNS = %d", *colp));
-	    }
-
 	    _nc_default_screensize(termp, linep, colp);
 	} else {
 	    _nc_check_screensize(sp, termp, linep, colp);
