--- a/third_party/sjpeg/CMakeLists.txt
+++ b/third_party/sjpeg/CMakeLists.txt
@@ -44,20 +44,7 @@
 ################################################################################
 # Android only.
 
-if(ANDROID)
-  include_directories(${SJPEG_ANDROID_NDK_PATH}/sources/android/cpufeatures)
-  add_library(cpufeatures-sjpeg
-    STATIC ${SJPEG_ANDROID_NDK_PATH}/sources/android/cpufeatures/cpu-features.c
-  )
-  target_link_libraries(cpufeatures-sjpeg dl)
-  set(SJPEG_DEP_LIBRARIES ${SJPEG_DEP_LIBRARIES} cpufeatures-sjpeg)
-  set(SJPEG_DEP_INCLUDE_DIRS ${SJPEG_DEP_INCLUDE_DIRS}
-      ${SJPEG_ANDROID_NDK_PATH}/sources/android/cpufeatures
-  )
-endif()
-
 ## Check for SIMD extensions.
-include(${CMAKE_CURRENT_LIST_DIR}/cmake/cpu.cmake)
 
 ################################################################################
 # sjpeg source files.
