diff -u -r ../cache/tmp-checkout/3rd/bee.lua/bee/utility/path_helper.cpp ./3rd/bee.lua/bee/utility/path_helper.cpp
--- ../cache/tmp-checkout/3rd/bee.lua/bee/utility/path_helper.cpp	2024-08-27 08:06:21.519466673 +0000
+++ ./3rd/bee.lua/bee/utility/path_helper.cpp	2024-08-27 08:10:38.315452462 +0000
@@ -102,6 +102,11 @@
         }
         return fs::path(argv[0]);
 #        else
+	char* termux_self_exe = getenv("TERMUX_EXEC__PROC_SELF_EXE");
+	if (termux_self_exe) {
+	    return fs::path(termux_self_exe);
+	}
+
         std::error_code ec;
         auto res = fs::read_symlink("/proc/self/exe", ec);
         if (ec) {
