diff -u -r ../clamav-1.3.1/libclamav/CMakeLists.txt ./libclamav/CMakeLists.txt
--- ../clamav-1.3.1/libclamav/CMakeLists.txt	2024-04-16 23:55:29.000000000 +0000
+++ ./libclamav/CMakeLists.txt	2024-05-20 13:27:59.862560548 +0000
@@ -446,7 +446,6 @@
         target_link_libraries( clamav
             PUBLIC
                 Threads::Threads
-                Iconv::Iconv
                 ${CMAKE_DL_LIBS}
                 m )
     endif()
@@ -463,8 +462,6 @@
 
     if(WIN32)
         set_target_properties( clamav PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON )
-    else()
-        target_link_libraries( clamav PUBLIC Iconv::Iconv )
     endif()
     if(UNIX AND NOT APPLE)
         target_link_options( clamav PRIVATE "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/libclamav.map")
@@ -559,7 +556,6 @@
         target_link_libraries( clamav_static
             PUBLIC
                 Threads::Threads
-                Iconv::Iconv
                 ${CMAKE_DL_LIBS}
                 m )
     endif()
