diff -u -r ../cache/tmp-checkout/make/modules/java.desktop/lib/Awt2dLibraries.gmk ./make/modules/java.desktop/lib/Awt2dLibraries.gmk
--- ../cache/tmp-checkout/make/modules/java.desktop/lib/Awt2dLibraries.gmk	2024-05-18 21:23:10.764706570 +0000
+++ ./make/modules/java.desktop/lib/Awt2dLibraries.gmk	2024-05-20 05:06:44.154740781 +0000
@@ -542,7 +542,7 @@
     LDFLAGS_unix := -L$(INSTALL_LIBRARIES_HERE), \
     LDFLAGS_aix := -Wl$(COMMA)-berok, \
     LIBS := $(BUILD_LIBFONTMANAGER_FONTLIB), \
-    LIBS_unix := -lawt -lawt_xawt -ljava -ljvm -lc $(LIBM) $(LIBCXX), \
+    LIBS_unix := -lawt -ljava -ljvm -lc $(LIBM) $(LIBCXX), \
     LIBS_macosx := -lawt_lwawt -framework CoreText -framework CoreFoundation -framework CoreGraphics, \
     LIBS_windows := $(WIN_JAVA_LIB) advapi32.lib user32.lib gdi32.lib \
         $(WIN_AWT_LIB), \
