diff -u -r ../bitcoin-27.0/depends/packages/bdb.mk ./depends/packages/bdb.mk
--- ../bitcoin-27.0/depends/packages/bdb.mk	2024-04-15 16:28:32.000000000 +0000
+++ ./depends/packages/bdb.mk	2024-05-14 21:16:24.081957818 +0000
@@ -13,7 +13,7 @@
 $(package)_config_opts_freebsd=--with-pic
 $(package)_config_opts_netbsd=--with-pic
 $(package)_config_opts_openbsd=--with-pic
-$(package)_config_opts_android=--with-pic
+$(package)_config_opts_android=--with-pic ac_cv_func_shmget=no
 $(package)_cflags+=-Wno-error=implicit-function-declaration -Wno-error=format-security -Wno-error=implicit-int
 $(package)_cppflags_freebsd=-D_XOPEN_SOURCE=600 -D__BSD_VISIBLE=1
 $(package)_cppflags_netbsd=-D_XOPEN_SOURCE=600
