diff -u -r ../command-not-found-2.4.0/command-not-found.cpp ./command-not-found.cpp
--- ../command-not-found-2.4.0/command-not-found.cpp	2023-11-27 06:08:59.000000000 +0000
+++ ./command-not-found.cpp	2024-05-23 12:18:34.160982200 +0000
@@ -39,34 +39,6 @@
 #endif
 };
 
-const std::list<std::string_view> root_commands = {
-#ifdef __aarch64__
-#include "commands-aarch64-termux-root.h"
-#elif defined __arm__
-#include "commands-arm-termux-root.h"
-#elif defined __i686__
-#include "commands-i686-termux-root.h"
-#elif defined __x86_64__
-#include "commands-x86_64-termux-root.h"
-#else
-#error Failed to detect arch
-#endif
-};
-
-const std::list<std::string_view> x11_commands = {
-#ifdef __aarch64__
-#include "commands-aarch64-termux-x11.h"
-#elif defined __arm__
-#include "commands-arm-termux-x11.h"
-#elif defined __i686__
-#include "commands-i686-termux-x11.h"
-#elif defined __x86_64__
-#include "commands-x86_64-termux-x11.h"
-#else
-#error Failed to detect arch
-#endif
-};
-
 struct info {
   std::string binary, repository;
 };
@@ -180,18 +152,6 @@
   if (res != 0) {
     return res;
   }
-
-  res = termux_look_for_packages(command, root_commands, &best_distance,
-                                 package_map, "root");
-  if (res != 0) {
-    return res;
-  }
-
-  res = termux_look_for_packages(command, x11_commands, &best_distance,
-                                 package_map, "x11");
-  if (res != 0) {
-    return res;
-  }
 
   if (best_distance == -1 || best_distance > 3) {
     std::cerr << command << ": command not found" << std::endl;
