diff -u -r ../command-not-found-2.4.0/generate-db.js ./generate-db.js
--- ../command-not-found-2.4.0/generate-db.js	2023-11-27 06:08:59.000000000 +0000
+++ ./generate-db.js	2024-05-23 12:14:35.380001688 +0000
@@ -7,7 +7,7 @@
 const gunzipAsync = promisify(gunzip);
 
 // TODO(@thunder-coding): Do not hardcode list of known architectures.
-const archs = ["aarch64", "arm", "i686", "x86_64"];
+const archs = ["aarch64", "x86_64"];
 const { TERMUX_PKG_CACHEDIR, TERMUX_PREFIX } = process.env;
 
 if (!TERMUX_PKG_CACHEDIR) {
