diff -u -r ../termux-tools-3.0.8/scripts/pkg.in ./scripts/pkg.in
--- ../termux-tools-3.0.8/scripts/pkg.in	2024-08-27 06:17:26.000000000 +0000
+++ ./scripts/pkg.in	2025-02-14 17:14:07.313518612 +0000
@@ -77,34 +77,10 @@
 }
 
 update_apt_cache() {
-	local current_host
-	current_host=$(head -n 1 <(sed -nE -e 's|^\s*deb\s+https?://(.+)\s+stable\s+main$|\1|p' @TERMUX_PREFIX@/etc/apt/sources.list) || true)
-
-	if [ -z "$current_host" ]; then
-		# No primary repositories configured?
-		apt update
-		return
-	fi
-
-	local metadata_file
-	metadata_file=$(
-		list_prefix=$(echo "$current_host" | sed 's|/|_|g')
-		arch=$(dpkg --print-architecture)
-		echo "@TERMUX_PREFIX@/var/lib/apt/lists/${list_prefix}_dists_stable_main_binary-${arch}_Packages" | sed 's|__|_|g'
-	)
-
-	if [ ! -e "@TERMUX_CACHE_DIR@/apt/pkgcache.bin" ] || [ ! -e "$metadata_file" ]; then
-		apt update
-		return
-	fi
-
 	local cache_modified
 	cache_modified=$(last_modified "@TERMUX_CACHE_DIR@/apt/pkgcache.bin")
 
-	local sources_modified
-	sources_modified=$(last_modified "@TERMUX_PREFIX@/etc/apt/sources.list")
-
-	if (( sources_modified <= cache_modified )) || (( cache_modified > 1200 )); then
+	if (( cache_modified > 1200 )); then
 		apt update
 	fi
 }
