diff -u -r ../samba-4.20.1/auth/credentials/credentials_krb5.c ./auth/credentials/credentials_krb5.c
--- ../samba-4.20.1/auth/credentials/credentials_krb5.c	2024-01-29 14:41:25.221069800 +0000
+++ ./auth/credentials/credentials_krb5.c	2024-05-16 22:09:51.931508803 +0000
@@ -626,7 +626,7 @@
 		must_free_cc_name = true;
 
 		if (lpcfg_parm_bool(lp_ctx, NULL, "credentials", "krb5_cc_file", false)) {
-			ccache_name = talloc_asprintf(ccc, "FILE:/tmp/krb5_cc_samba_%u_%p",
+			ccache_name = talloc_asprintf(ccc, "FILE:@TERMUX_PREFIX@/tmp/krb5_cc_samba_%u_%p",
 						      (unsigned int)getpid(), ccc);
 		} else {
 			ccache_name = talloc_asprintf(ccc, "MEMORY:%p",
