diff -u -r ../cache/tmp-checkout/make/autoconf/libraries.m4 ./make/autoconf/libraries.m4
--- ../cache/tmp-checkout/make/autoconf/libraries.m4	2024-05-18 21:23:10.561707370 +0000
+++ ./make/autoconf/libraries.m4	2024-05-18 21:44:24.581666149 +0000
@@ -48,7 +48,7 @@
   else
     # All other instances need X11, even if building headless only, libawt still
     # needs X11 headers.
-    NEEDS_LIB_X11=true
+    NEEDS_LIB_X11=false
   fi
 
   # Check if fontconfig is needed
