diff -u -r ../libspatialite-5.1.0/src/gaiageo/gg_shape.c ./src/gaiageo/gg_shape.c
--- ../libspatialite-5.1.0/src/gaiageo/gg_shape.c	2023-08-04 07:51:40.000000000 +0000
+++ ./src/gaiageo/gg_shape.c	2024-05-16 20:55:57.619067357 +0000
@@ -75,7 +75,7 @@
 #include <localcharset.h>
 #endif /* end localcharset */
 #else /* not MINGW32 */
-#if defined(__APPLE__) || defined(__ANDROID__)
+#if defined(__APPLE__)
 #include <iconv.h>
 #include <localcharset.h>
 #else /* neither Mac OsX nor Android */
