diff -uNr screen-4.7.0/setuid.c screen-4.7.0.mod/setuid.c
--- screen-4.7.0/setuid.c	1970-01-01 03:00:00.000000000 +0300
+++ screen-4.7.0.mod/setuid.c	2019-10-14 02:14:35.810387228 +0300
@@ -0,0 +1,25 @@
+#include <pwd.h>
+
+int setuid(uid_t uid) {
+	return 0;
+}
+
+int setgid(gid_t gid) {
+	return 0;
+}
+
+int seteuid(gid_t gid) {
+	return 0;
+}
+
+int setegid(gid_t gid) {
+	return 0;
+}
+
+int setresuid(uid_t ruid, uid_t euid, uid_t suid) {
+	return 0;
+}
+
+int setresgid(gid_t rgid, gid_t egid, gid_t sgid) {
+	return 0;
+}
