diff -u -r ../go/src/os/executable_procfs.go ./src/os/executable_procfs.go
--- ../go/src/os/executable_procfs.go	2025-02-10 23:33:55.000000000 +0000
+++ ./src/os/executable_procfs.go	2025-02-23 09:53:37.081649355 +0000
@@ -17,8 +17,10 @@
 	switch runtime.GOOS {
 	default:
 		return "", errors.New("Executable not implemented for " + runtime.GOOS)
-	case "linux", "android":
+	case "linux":
 		procfn = "/proc/self/exe"
+	case "android":
+		return runtime.AndroidSelfExecutable(), nil
 	}
 	path, err := Readlink(procfn)
 
