diff -u -r ../traceroute-2.1.5/traceroute/mod-tcp.c ./traceroute/mod-tcp.c
--- ../traceroute-2.1.5/traceroute/mod-tcp.c	2023-12-13 04:05:57.000000000 +0000
+++ ./traceroute/mod-tcp.c	2024-05-14 12:42:18.412318979 +0000
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <string.h>
 #include <sys/socket.h>
 #include <poll.h>
 #include <netinet/icmp6.h>
diff -u -r ../traceroute-2.1.5/traceroute/mod-tcpconn.c ./traceroute/mod-tcpconn.c
--- ../traceroute-2.1.5/traceroute/mod-tcpconn.c	2023-12-09 02:50:31.000000000 +0000
+++ ./traceroute/mod-tcpconn.c	2024-05-14 12:41:58.337595474 +0000
@@ -8,6 +8,7 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 #include <sys/socket.h>
 #include <poll.h>
 #include <netinet/icmp6.h>
diff -u -r ../traceroute-2.1.5/traceroute/traceroute.c ./traceroute/traceroute.c
--- ../traceroute-2.1.5/traceroute/traceroute.c	2023-12-10 04:27:41.000000000 +0000
+++ ./traceroute/traceroute.c	2024-05-14 12:42:34.159101141 +0000
@@ -11,6 +11,7 @@
 #include <stdarg.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <string.h>
 #include <sys/socket.h>
 #include <poll.h>
 #include <netinet/icmp6.h>
